core/actor/src/main/scala/net/liftweb/actor/LAFuture.scala

```diff
@@ -406,5 +408,28 @@ object LAFuture {
 
     ret
   }
-}
 
+  private def inContext[T](f: () => T, context: Box[LAFuture.Context]): () => T = {
+    context.map(_.around(f)) openOr f
+  }
+
+  private def inContext[A, T](f: (A) => T, context: Box[LAFuture.Context]): (A) => T = {
+    context.map(_.around(f)) openOr f
+  }
+
+  /**
+    * Allows to wrap function in another function providing some additional functionality.
+    * It may choose to execute or not execute that functionality, but should not interpret
+    * or change the returned value; instead, it should perform orthogonal actions that
+    * need to occur around the given functionality. Typical example is setting up DB
+    * transaction.
```

Comment:
I'm not a native speaker, so any improvement ideas for this scaladoc are very appreciated :) 

As an "orthogonal" action, such as DB transaction I mean:

```
def addNewUser(name: String, surname: String, email: String): Box[User] = {
  try {
    DB.createNewTransaction()
    val user = User(name, surname, email)
    repository.persist(user)
    eventBus.publish(NewUserCreatedEvent(user.id))
 } finally {
   DB.flushTransaction()
 }
} 
```

^ Here, that `try - finally` block with all `DB` calls is orthogonal to business logic of this method. It shouldn't be here together with business logic code. That's a typical case when we can use `Context` (or `CommonLoanWrapper`). In Java world, it would be probably just an annotation +  aspect.